config-model/src/main/java/com/yahoo/schema/OnnxModel.java

```diff
@@ -39,6 +39,19 @@ public OnnxModel(String name, String fileName) {
         validate();
     }
 
+    @Override
+    public OnnxModel clone() {
+        try {
+            OnnxModel clone = (OnnxModel) super.clone();
```

Comment:
This will be a shallow clone but I think you want a deep one?
So, you should initialize inputMap, outputMap and initializers to new, empty collections before put/addAll, and deep clone modelInfo.

Reply:
Actually, a shallow clone would work here, because the `inputMap`, `outputMap`, `initializers` and `modelInfo` are model-specific parameters, while only the last 4 are configurable via `services.xml` and thus cluster-specific. However, that would probably break the clone contract?

Reply:
It's ok then but then you shouldn't add anything to those existing collections in clone.

And consider if this may lead to bugs later, maybe either deep clone anyway, or separate the general model and use-specific values into two classes.

Reply:
Removed the part of adding to the existing collections. Looking at the code, the fields are effectively immutable so specified that shallow clone is sufficient for now.